jormungandr/src/blockchain/chain.rs

```diff
@@ -63,61 +63,55 @@ use chain_time::TimeFrame;
 use std::sync::Arc;
 use tokio_stream::StreamExt;
 
-// derive
-use thiserror::Error;
-
-error_chain! {
-    foreign_links {
-        Storage(StorageError);
-        Ledger(ledger::Error);
-        Block0(Block0Error);
-    }
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error(transparent)]
+    Storage(#[from] StorageError),
 
-    errors {
-        Block0InitialLedgerError {
-            description("Error while creating the initial ledger out of the block0")
-        }
+    #[error(transparent)]
+    Ledger(#[from] ledger::Error),
 
-        Block0AlreadyInStorage {
-            description("Block0 already exists in the storage")
-        }
+    #[error(transparent)]
+    Block0(#[from] Block0Error),
 
-        Block0NotAlreadyInStorage {
-            description("Block0 is not yet in the storage")
-        }
+    #[error(transparent)]
+    Intercom(#[from] crate::intercom::Error),
 
-        MissingParentBlock(hash: HeaderHash) {
-            description("missing a parent block from the storage"),
-            display(
-                "Missing a block from the storage. The node was recovering \
-                 the blockchain and the parent block '{}' was not \
-                 already stored",
-                hash,
-            ),
-        }
+    #[error(transparent)]
+    SendError(#[from] futures::channel::mpsc::SendError),
 
-        NoTag (tag: String) {
-            description("Missing tag from the storage"),
-            display("Tag '{}' not found in the storage", tag),
-        }
+    #[error("failed to send message to the fragments process")]
+    TransactionSendError(#[from] crate::async_msg::TrySendError<crate::intercom::TransactionMsg>),
 
-        BlockHeaderVerificationFailed (reason: String) {
-            description("Block header verification failed"),
-            display("The block header verification failed: {}", reason),
-        }
+    #[error("Error while creating the initial ledger out of the block0")]
```

Comment:
Nit: the prevailing convention (not everywhere in our code, though) is to start error messages with lower case.